client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import React, { useMemo, useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+import classNames from 'classnames';
+import { ToggleControl } from '@wordpress/components';
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	JETPACK_PRODUCTS_BY_TERM,
+	PRODUCT_JETPACK_BACKUP_REALTIME,
+} from 'calypso/lib/products-values/constants';
+import {
+	JETPACK_RESET_PLANS_BY_TERM,
+	PLAN_JETPACK_SECURITY_REALTIME,
+} from 'calypso/lib/plans/constants';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { TERM_MONTHLY, TERM_ANNUALLY } from 'calypso/lib/plans/constants';
+import { masterbarIsVisible } from 'calypso/state/ui/selectors';
+import { getJetpackCROActiveVersion } from '../abtest';
+import useDetectWindowBoundary from '../use-detect-window-boundary';
+import { getHighestAnnualDiscount } from '../utils';
+
+/**
+ * Type dependencies
+ */
+import type { FilterBarProps, DiscountMessageProps } from '../types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const CALYPSO_MASTERBAR_HEIGHT = 47;
+const CLOUD_MASTERBAR_HEIGHT = 94;
+
+const REALTIME_PRODUCTS = [ PLAN_JETPACK_SECURITY_REALTIME, PRODUCT_JETPACK_BACKUP_REALTIME ];
+
+const DiscountMessage: React.FC< DiscountMessageProps > = ( { primary } ) => {
+	const translate = useTranslate();
+	const isMobile: boolean = useMobileBreakpoint();
+
+	const croVersion = getJetpackCROActiveVersion();
+	const slugsToCheck = useMemo(
+		() =>
+			[
+				...JETPACK_PRODUCTS_BY_TERM.map( ( s ) => s.yearly ),
+				...JETPACK_RESET_PLANS_BY_TERM.map( ( s ) => s.yearly ),
+			].filter( ( slug ) => {
+				// Don't factor in real-time products for CRO v1 --
+				// they're not visible in the product grid
+				if ( croVersion === 'v1' && REALTIME_PRODUCTS.includes( slug ) ) {
+					return false;
+				}
+
+				return true;
+			} ),
+		[ croVersion ]
+	);
+	const highestAnnualDiscount = useSelector( ( state ) =>
+		getHighestAnnualDiscount( state, slugsToCheck )
+	);
+
+	if ( ! highestAnnualDiscount ) {
+		return null;
+	}
+
+	return (
+		<div className={ classNames( 'plans-filter-bar-i5__discount-message', { primary: primary } ) }>
+			<div>
+				<span className="message">
+					{ isMobile
+						? translate( 'SAVE %(discount)s BY PAYING YEARLY', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( 'Save %(discount)s by paying yearly' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/plans-filter-bar-i5)